packages/teams-js/src/public/meeting.ts

```diff
@@ -613,4 +655,125 @@ export namespace meeting {
+   * set isAppHandlingAudio=false in params, will restore the Host(Teams) audio
+   *   Deregister the mic mute status change events and App will not be receive the events.
+   *
+   * @throws standard parameter validation error
+   * @param requestAppAudioHandlingParams - RequestAppAudioHandlingParams interface, and contains
+   *   isAppHandlingAudio - true - expect App to handle the audio and Teams to go audioless.
+   *     false - Audio transferred to Teams and Teams to restore the audio.
+   *   callbackMicMuteStateChangedHandler - Callback contains mic status param.
+   * @param callback - Callback contains error and operation result parameters.
+   * error can either contain an error of type SdkError (error indication), or null (non-error indication)
+   * result can either contain a true boolean value (successful termination), or null (unsuccessful fetch)
+   *
+   * @beta
+   */
+  export function requestAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (isHostAudioless: boolean) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[requestAppAudioHandling] Callback response cannot be null');
+    }
+    if (!requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler) {
+      throw new Error('[requestAppAudioHandling] Callback Mic mute state handler cannot be null');
+    }
+    ensureInitialized(runtime, FrameContexts.sidePanel, FrameContexts.meetingStage);
+
+    if (requestAppAudioHandlingParams.isAppHandlingAudio) {
+      startAppAudioHandling(requestAppAudioHandlingParams, callback);
+    } else {
+      stopAppAudioHandling(requestAppAudioHandlingParams, callback);
+    }
+  }
+
+  function startAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (isHostAudioless: boolean) => void,
+  ): void {
+    const callbackInternalRequest = (error: SdkError | null, isHostAudioless: boolean | null): void => {
+      if (error && isHostAudioless != null) {
+        throw new Error('[requestAppAudioHandling] Callback response - both parameters cannot be set');
+      }
+      if (error) {
+        throw new Error(`[requestAppAudioHandling] Callback response - SDK error ${error.errorCode} ${error.message}`);
+      }
+      if (typeof isHostAudioless !== 'boolean') {
+        throw new Error('[requestAppAudioHandling] Callback response - isHostAudioless must be a boolean');
+      }
+
+      const micStateChangedCallback = async (micState: MicState): Promise<void> => {
+        try {
+          const newMicState = await requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler(micState);
+
+          const micStateDidUpdate = newMicState.isMicMuted === micState.isMicMuted;
+          setMicStateWithReason(
+            newMicState,
+            micStateDidUpdate ? MicStateChangeReason.HostInitiated : MicStateChangeReason.AppDeclinedToChange,
```

Comment:
I think you have this backwards, right now micStateDidUpdate is true if the value of isMicMuted didn't change. (see line 715). Probably you want the opposite of that for this code to make sense.

Reply:
(Assuming I'm correct about this, I would have also expected a unit test to catch this and fail. Not sure if that means you didn't write a unit test for this case or if something else is missing, but would be good to look into)

Reply:
I believe this is correct because when `micState` and `newMicState` have the same value, that means the app has applied the mic state that was requested from the host. If it returned a different value, that means the value the host requested is not reflected in the app

Reply:
Ah okay gotcha, that particular detail I missed but makes sense